core/src/toga/icons.py

```diff
@@ -1,13 +1,17 @@
 from __future__ import annotations
 
+import sys
 from pathlib import Path
 from typing import TYPE_CHECKING
 
 import toga
 from toga.platform import get_platform_factory
 
 if TYPE_CHECKING:
-    from typing import TypeAlias
+    if sys.version_info < (3, 10):
+        from typing_extensions import TypeAlias
+    else:
+        from typing import TypeAlias
```

Comment:
You can leave as-is if you want (since I'll have to handle all this in #2252) but you don't have to worry about a conditional import. `pyupgrade` will automatically switch to using only `typing` to import `TypeAlias` once 3.9 is no longer supported.